_posts/2016-06-14-https-securise-websites.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: post
+title:  "How Etsy France switched all its sites to HTTPS"
+date:   2016-07-15
+categories: projects
+tags: [security, https, http, migration, project]
+author: antoine
+---
+
+When Etsy acquired A little Market, the migration of our sites to https which was on the shelf since a few years became one of our top priority projects in mid 2015.  
+But before starting such a big and impacting project, we went through a long preparation phase.
+
+## Project preparation
+
+### Why ?
+
+The goal of this project was to force all our users to access our services through HTTPS connections. Google and other search engines would have to crawl our websites only in HTTPS to avoid duplicate content.
+
+### How
+
+We want to switch smoothly and easily from HTTP to HTTPS. At most, we want to update a few configuration values to do it progressively.
+All the blocking points to use HTTPS will have to be fixed before the deadline.
+
+### Acceptance criteria
+
+All our applications must be accessible through https.  
+All users accessing the websites through http must be redirected to https using 301 (301 redirections are very important for Search engines).  
+
+### When to switch ?
+
+A switch from HTTP to HTTPS can have a big impact on search engine's ranking. As we do not use paid marketing, a bad indexing in search engines and especially Google can have really bad consequences on our traffic. Thanks to previous experiences, our SEO consultant estimated that SEO traffic returns back to normal around 3 months after switching.  
+Because we had several marketing operations during the year (winter and summer sales, private sales, Christmas, etc.) and we didn't want a big impact on our SEO traffic during these operations, we carefully determined the periods when the switch could be made. We only found two:
+
+- February/March (traffic should return back to normal around May)
+- July/August (traffic should return back to normal around October)
+
+
+## Milestones
+
+1. Dependencies
+    1. List all the dependencies of our main applications (ex: assets, blogs, images, etc)
+    2. These dependencies will have to be migrated first.
+
+2. We know some of our hostnames contain more than one subdomain level (ex: static.commun.alittlemarket.com) but we don't want to buy expensive certificates that cover every level.
+    1. List all the concerned hostnames
+    2. Fix incompatible subdomains (ex: static.commun.alittlemarket.com -> static-commun.alittlemarket.com). Fix them means:
+        1. Create new subdomains in apache
+        2. Add them in [Route53](https://aws.amazon.com/route53/)
```

Comment:
maybe a small high level infrastructure diagram would be nice (you're talking about AWS just now, aren't you?)